ktor-client/ktor-client-plugins/ktor-client-content-negotiation/ktor-client-content-negotiation-tests/jvm/src/io/ktor/client/plugins/contentnegotiation/tests/JsonContentNegotiationTest.kt

```diff
@@ -231,4 +231,58 @@ abstract class JsonContentNegotiationTest(private val converter: ContentConverte
             assertEquals("OK", it.bodyAsText())
         }
     }
+
+    @Test
+    fun testRespondWithTypeInfoAny() = testApplication {
+        install(ContentNegotiation) {
+            register(ContentType.Application.Json, converter)
+        }
+
+        fun buildResponse(): Any = Wrapper("abc")
+
+        routing {
+            get("/") {
+                call.respond(buildResponse())
+            }
+        }
+
+        createClient {
+            install(io.ktor.client.plugins.contentnegotiation.ContentNegotiation) {
+                register(ContentType.Application.Json, converter)
+            }
+        }.get("/").let { response ->
+            assertEquals(HttpStatusCode.OK, response.status)
+            assertEquals("""{"value":"abc"}""", response.bodyAsText())
+        }
+    }
+
+    @Test
+    fun testRespondSealedWithTypeInfoAny() = testApplication {
+        install(ContentNegotiation) {
+            register(ContentType.Application.Json, converter)
+        }
+
+        fun buildResponse(): Any = DataType("abc")
+
+        routing {
+            get("/") {
+                call.respond(buildResponse())
+            }
+        }
+
+        createClient {
+            install(io.ktor.client.plugins.contentnegotiation.ContentNegotiation) {
+                register(ContentType.Application.Json, converter)
+            }
+        }.get("/").let { response ->
+            assertEquals(HttpStatusCode.OK, response.status)
+            assertEquals("""{"value":"abc"}""", response.bodyAsText())
+        }
+    }
 }
+
+@Serializable
+internal sealed class Data
```

Comment:
can you also add a test with serialisation of a sub type?

Reply:
sure